docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+- can be combined with other isolation methods
+
+### Disadvantages:
+- a single Kubernetes cluster oriented, cross clusters restrictions can be based on IP blocks only, which significantly 
+complicates the configuration. Therefore, later we will provide other isolation methods as an extension to NP.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Kind deployments which by default use "kindnetd" do not
+support NP. Support NP requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). On the other hand, all K8s 
+clusters in production deployments use CNIs which support NP.
+
+## Implementation
+
+The `from` element of `ingress` 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on `podSelectors` (labels based), and/or 
+`namespaceSelectors` (labels based) and/or `ipBlocks`. The last one permits incoming network traffic from predefined IP 
+address blocks. 
+The `to` element of `egress` restricts outgoing connections and has similar structure. 
+
+Current `FybrikApplicationSpec` has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of `clusterName` and `workloadSelector`. Unfortunately, based on this information we can 
+implement only one label-based `podSelector`, which will allow incoming connections from pods in the namespace where the NP 
+instance is deployed and have the defined label. In order to restrict incoming connections to Fybrik modules, the NP
+instances should be deployed in teh same namespace where the modules are running. However, Fybrik separates user's 
+objects, e.g. workloads and Fybrik objects, e.g. modules. Therefore, the existing mechanism should be extended.
+
+### Suggestions:
+
+- Extend the `FybrikApplicationSpec.selector` element with an array of `WorkloadLocations`, this array will be an input 
+element for the Network Policy `from`element of the data plane entry module. 
+- The `WorkloadLocation` wil have the following structure: 
+
+```go
+type WorkloadLocation struct {
+    // This is a workload's pod label selector which selects user workload pods. This field follows standard label
+    // selector semantics; if present but empty, it selects all pods.
+    //
+    // If NamespaceSelector is also set, then the WorkloadLocation as a whole selects
+    // the workload pods matching WorkloadPodSelector in the namespaces selected by NamespaceSelector.
+    // Otherwise, it selects the pods matching WorkloadPodSelector in the Fybrik data plane entry point namespace.
```

Comment:
According to this semantics, I would assume that if NamespaceSelector is not set, all namespaces are selected.